MapleServer2/Servers/Game/FieldManager.cs

```diff
@@ -396,5 +386,39 @@ private Task StartHealingSpot(GameSession session, IFieldObject<Player> player)
                 }
             });
         }
+
+        private void SpawnMobs(IFieldObject<MobSpawn> mobSpawn)
+        {
+            List<CoordF> spawnPoints = MobSpawn.SelectPoints(mobSpawn.Value.SpawnRadius);
+
+            foreach (NpcMetadata mob in mobSpawn.Value.SpawnMobs)
+            {
+                int spawnCount = mob.NpcMetadataBasic.GroupSpawnCount;  // Spawn count changes due to field effect (?)
+                if (spawnCount > mobSpawn.Value.MaxPopulation)
+                {
+                    break;
+                }
+
+                for (int i = 0; i < spawnCount && i < mobSpawn.Value.MaxPopulation; i++)
```

Comment:
Because of the check on line 397, why are both of these i checks needed? `spawnCount` should always be less meaning the second check seem useless to me?